app/controllers/admin/users_controller.rb

```diff
@@ -117,8 +117,12 @@ def delete_project_proposal
   end
 
   def delete_user
-    User.find(params[:id]).destroy
-    redirect_to root_path, notice: 'User Deleted!'
+    if User.authenticate(params[:username_email], params[:password]).admin?
```

Comment:
I would change this line to `if @user.admin? && @user.pword == password`. The reason being that this way the email doesn't need to be re-inputed.